modules/javafx.graphics/src/main/java/com/sun/glass/ui/WindowControlsOverlay.java

```diff
@@ -0,0 +1,540 @@
+ *
+ * <h2>Substructure</h2>
+ * <ul>
+ *     <li>{@link Region} — {@code window-button-container}
+ *     <ul>
+ *         <li>{@link Region} — {@code window-button}, {@code minimize-button}
+ *         <li>{@link Region} — {@code window-button}, {@code maximize-button}
+ *         <li>{@link Region} — {@code window-button}, {@code close-button}
+ *     </ul>
+ * </ul>
+ *
+ * <table style="white-space: nowrap">
+ *     <caption>CSS properties of {@code window-button-container}</caption>
+ *     <thead>
+ *         <tr><th>CSS property</th><th>Values</th><th>Default</th><th>Comment</th></tr>
+ *     </thead>
+ *     <tbody>
+ *         <tr>
+ *             <th>-fx-button-placement</th><td>[ left | right ]</td><td>right</td>
+ *             <td style="white-space: break-line">
+ *                 Specifies the placement of the window buttons on the left or the right side of the window.
+ *             </td>
+ *         </tr>
+ *         <tr>
+ *             <th>-fx-allow-rtl</th><td>&lt;boolean&gt;</td><td>true</td>
+ *             <td style="white-space: break-line">
+ *                 Specifies whether the minimize/maximize/close buttons support right-to-left orientations.
+ *             </td>
+ *         </tr>
+ *     </tbody>
+ * </table>
+ *
+ * <table style="white-space: nowrap">
+ *     <caption>CSS properties of {@code minimize-button}, {@code maximize-button}, {@code close-button}</caption>
+ *     <thead>
+ *         <tr><th>CSS property</th><th>Values</th><th>Default</th><th>Comment</th></tr>
+ *     </thead>
+ *     <tbody><tr>
+ *         <th>-fx-button-order</th><td>&lt;integer&gt;</td><td>0/1/2</td>
+ *         <td style="white-space: break-line">
+ *             Specifies the layout order of a button relative to the other buttons.
+ *             Lower values are laid out before higher values.</td>
+ *     </tr></tbody>
+ * </table>
+ *
+ * <table style="white-space: nowrap">
+ *     <caption>Conditional style classes of window buttons</caption>
+ *     <thead>
+ *         <tr style="min-width: 200"><th>Style class</th><th>Applies to</th><th>Comment</th></tr>
+ *     </thead>
+ *     <tbody>
+ *         <tr><th>.dark</th>
+ *             <td>all buttons</td>
+ *             <td style="white-space: break-line">
+ *                 This style class will be present if the brightness of {@link Scene#fillProperty()}
+ *                 as determined by {@link Utils#calculateAverageBrightness(Paint)} is less than 0.5
+ *             </td>
+ *         </tr>
+ *         <tr><th>.restore</th><th>{@code maximize-button}</th><td style="white-space: break-line">
+ *             This style class will be present if {@link Stage#isMaximized()} is {@code true}</td>
+ *         </tr>
+ *     </tbody>
+ * </table>
+ */
+public final class WindowControlsOverlay extends Region {
+
+    private static final CssMetaData<WindowControlsOverlay, HorizontalDirection> BUTTON_PLACEMENT_METADATA =
```

Comment:
BTW, check out `StyleablePropertyFactory`.  It can be used if one doesn't do lazy initialization, or if one doesn't need e.g. `IntegerProperties`.